pkg/cmd/cli/cmd/importimage.go

```diff
@@ -374,13 +374,15 @@ func (o *ImportImageOptions) createImageImport() (*imageapi.ImageStream, *imagea
 		},
 		Spec: imageapi.ImageStreamImportSpec{Import: true},
 	}
+	insecureAnnotation := stream.Annotations[imageapi.InsecureRepositoryAnnotation]
+	insecure := o.Insecure || insecureAnnotation == "true"
```

Comment:
For why not - no opinions. For why yes - because that's what we do everywhere else, the flag has higher precedence than what's already there. I'll prepare an update to that...